pypeit/images/rawimage.py

```diff
@@ -653,6 +680,8 @@ def process(self, par, bpm=None, scattlight=None, flatimages=None, bias=None, sl
         flat_bpm = self.flatfield(flatimages, slits=slits, debug=debug) if self.use_flat else None
 
         # Calculate the inverse variance
+        # TODO :: I think the IVAR should technically be calculated before the flat-fielding.
+        #         This is because the flat-fielding is done in counts, and the IVAR is in counts^2.
```

Comment:
I'm not sure I follow.  The flat-fielding is a scale factor that is applied to the counts in the image.  `build_ivar` incorporates this scale factor in the IVAR calculation.  I.e., the IVAR is in ("corrected counts")^2, not counts^2.  Happy to discuss more though.